### PR TITLE
Remove _basis from output arguments

### PR DESCRIPTION
**Pull request by vlaaad**
This commit is related to #5447. It removes basis from node output fnk's
arguments, which is a prerequisite for a different caching strategy that
will eventually solve the issue.

### CODE REVIEW

File: editor/test/integration/subselection_test.clj [hunk 1]
```diff
@@ -133,8 +133,8 @@
 
 (g/defnode MoveManip
   (input selection g/Any)
-  (output position g/Any (g/fnk [selection _basis]
-                                (let [evaluation-context (g/make-evaluation-context {:basis _basis})
+  (output position g/Any (g/fnk [selection]
+                                (let [evaluation-context (g/make-evaluation-context)
```
Comment (matgis) [inline editor/test/integration/subselection_test.clj:137]: Whoa. Super-weird that we're evaluating `g/node-value` inside an output. We better hope that no-one caches this downstream, or we'll run into trouble. 😨

**Review by matgis — COMMENTED** [review]
I'm pretty sure this would work. However, looking over the change, I'm tempted to not introduce the `_type` intrinsic and instead add `node-type` as a field to the `OverrideNode` record, like we have in `NodeImpl`. This would allow us to remove the `basis` argument from `gt/node-type`, and enable us to use the existing `_this` intrinsic to obtain the node type. I wonder if the recursive nature of the `gt/node-type` implementation in `OverrideNode` might be introducing quite a bit of overhead right now, seeing as how the node type is used internally in a lot of places. What do you think?

File: editor/test/integration/subselection_test.clj (hunk 1 above)
Comment (vlaaad) [reply editor/test/integration/subselection_test.clj:137]: Yeah, that's wild. Good thing it's a test and not a production code I guess...

**Comment by vlaaad** [comment]
I agree that putting node type into override node will simplify things, I'll try to do it!

File: editor/src/clj/dynamo/graph.clj [hunk 2]
```diff
@@ -83,15 +83,15 @@
...
 
 (defn node-type
   "Return the node-type given a node. Uses the current basis if not provided."
-  ([node]
-    (node-type (now) node))
-  ([basis node]
-    (when node
-      (gt/node-type node basis))))
+  [node]
+  (gt/node-type node))
```
Comment (matgis) [inline editor/src/clj/dynamo/graph.clj:91]: Presumably we should keep the `(when node ...)` clause here to preserve the original behavior? Or are we sure this never gets called with a `nil` node?

**Review by matgis — CHANGES_REQUESTED** [review]
Do we need to support `nil` values for `g/node-type` like we used to? Looks good otherwise!

File: editor/src/clj/dynamo/graph.clj (hunk 2 above)
Comment (vlaaad) [reply editor/src/clj/dynamo/graph.clj]: Good catch, fixed!

**Review by matgis — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**